src/braket/timings/time_series.py

```diff
@@ -87,6 +87,120 @@ def _ensure_sorted(self) -> None:
+
+        Returns:
+            TimeSeries: time series constructed from lists
+        """
+        if len(times) != len(values):
+            raise ValueError(
+                f"The lengths of the times({len(times)})\
+                      and values({len(values)}) lists are not equal."
+            )
+
+        ts = TimeSeries()
+        for t, v in zip(times, values):
+            ts.put(t, v)
+        return ts
+
+    @staticmethod
+    def constant_like(times: Union[List[float], TimeSeries], constant: float = 0.0) -> TimeSeries:
+        """Obtain a constant time series given the list of time points and the constant values
+
+        Args:
+            times (List[float]): list of time points
+
+        Returns:
+            TimeSeries: A constant time series
+        """
+        ts = TimeSeries()
+        for t in times:
+            ts.put(t, constant)
+        return ts
+
+    def concatenate(self, other: TimeSeries) -> TimeSeries:
+        """Concatenate two time series to a single time series
+
+        Args:
+            other (TimeSeries): The second time series to be concatenated
+
+        Returns:
+            TimeSeries: The concatenated time series.
+
+        Notes:
+            Keeps the time points in both time series unchanged.
+            Assumes that the time points in the first TimeSeries
+            are at earler times then the time points in the second TimeSeries.
+        """
+
+        if len(other.times()) * len(self.times()) == 0:
+            pass
+        elif not min(other.times()) > max(self.times()):
+            raise ValueError(
+                "The time points in the first TimeSeries must be strictly smaller \
+                then the time points in the second TimeSeries."
+            )
+
+        new_time_series = TimeSeries()
+        new_times = self.times() + other.times()
+        new_values = self.values() + other.values()
+        for t, v in zip(new_times, new_values):
+            new_time_series.put(t, v)
+
+        return new_time_series
+
+    def stitch(self, other: TimeSeries, boundary: str = "mean") -> TimeSeries:
+        """Stitch two time series to a single time series and shifts the time points accordingly.
+
+        Args:
+            other (TimeSeries): The second time series to be concatenated
+            boundary (str): {"mean", "left", "right"}. Boundary point handler.
+                Possible options are
+                    * "mean" - take the average of the boundary value points of the first
+                    and the second time series.
+                    * "left" - use the last value from the left time series as the boundary point.
+                    * "right" - use the first value from the right time series as the boundary
+                    point.
+        Returns:
+            TimeSeries: The stitched time series.
+        """
+
+        if len(self.times()) == 0:
+            return TimeSeries.from_lists(times=other.times(), values=other.values())
```

Comment:
Does this still make sense for "mean" and "left"?

Reply:
Yes, it makes sense. When the self.time_series is empty we simply return the second time series regardless of the boundary condition.  When we are stitching an empty and a non-empty time series, we are returning the non-empty one, since it is the only logical possibility.